### PR TITLE
Adding ImGui Support

### PR DESCRIPTION
**Pull request by wangra-google**
- using one extra composition layer Quad for the ImGui UI elements.

### CODE REVIEW

File: include/ppx/application.h [hunk 1]
```diff
@@ -242,6 +242,14 @@ struct ApplicationSettings
             grfx::Format depthFormat = grfx::FORMAT_UNDEFINED;
             uint32_t     imageCount  = 2;
         } swapchain;
+
+#if defined(PPX_BUILD_XR)
+        struct
+        {
+            XrVector3f  pos;
+            XrExtent2Df size;
+        } ui;
```
Comment (vettoreldaniele) [inline include/ppx/application.h:251]: Can you add a comment on what these coordinates represent? E.g. is `pos` the upper-left corner, center point, or something else?

File: include/ppx/application.h [hunk 2]
```diff
@@ -427,6 +440,8 @@ class Application
 #if defined(PPX_BUILD_XR)
     XrComponent mXrComponent;
     uint32_t    mDebugCaptureSwapchainIndex = 0;
+    uint32_t    mUISwapchainIndex           = 0;
+    uint32_t    mStereoScopicSwapchainIndex = 0;
```
Comment (vettoreldaniele) [inline include/ppx/application.h:444]: nit: stereoscopic is a single word, so probably should be  `mStereoscopicSwapchainIndex`

File: projects/04_cube_xr/main.cpp [hunk 3]
```diff
@@ -234,16 +245,53 @@ void ProjApp::Setup()
...
-    uint32_t currentViewIndex = 0;
+    PerFrame& frame            = mPerFrame[0];
+    uint32_t  imageIndex       = UINT32_MAX;
+    uint32_t  currentViewIndex = 0;
     if (GetSettings()->enableXR) {
         currentViewIndex = GetXrComponent().GetCurrentViewIndex();
     }
 
-    grfx::SwapchainPtr swapchain = GetSwapchain(currentViewIndex);
+    // render UI into a different composition layer
```
Comment (vettoreldaniele) [inline projects/04_cube_xr/main.cpp:255]: nit: comment formatting (capitalization, end period). Same for other comments

**Review by vettoreldaniele — CHANGES_REQUESTED** [review]
LGTM, just a few nits! Thanks for adding UI rendering support.

File: include/ppx/application.h (hunk 1 above)
Comment (wangra-google) [reply include/ppx/application.h:255]: fixed

File: include/ppx/application.h (hunk 2 above)
Comment (wangra-google) [reply include/ppx/application.h]: fixed

File: projects/04_cube_xr/main.cpp (hunk 3 above)
Comment (wangra-google) [reply projects/04_cube_xr/main.cpp]: fixed

**Review by vettoreldaniele — APPROVED** [review]

**Pull request merged**